security/subsystem/src/main/java/org/jboss/as/security/service/SimpleSecurityManager.java

```diff
@@ -352,7 +352,7 @@ public void push(final String securityDomain, String userName, char[] password,
         contexts.push(previous);
         SecurityContext current = establishSecurityContext(securityDomain);
         if (propagate &&  previous != null) {
-            current.setSubjectInfo(getSubjectInfo(previous));
+            propagateSubject(current,previous);
             current.setIncomingRunAs(previous.getOutgoingRunAs());
```

Comment:
As a full method has been added handling the propagation from a previous context to a current context shouldn't this second "copy" also be within the "propagateSubject" method?

Reply:
There are different ways of handling incoming identity, did not want to touch it. Fewer changes the better.